src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+                throw new EmptyDescriptionException(userInput[0]);
+            }
+            System.out.println(format);
+            taskArrayList.add(new ToDo (userInput[1]));
+            //count = taskAdder(taskArrayList, count);
+            taskAdder(taskArrayList);
+            System.out.println(format);
+            try {
+                writeFile();
+            } catch (IOException | DukeException e) {
+                System.out.println("There was a problem saving your task! Please try again!");
+                System.out.println(format);
+            }
+        }
+
+        else if (userInput[0].equals("deadline")){ //ADDING A DEADLINE
+            if (mainTask == null){
+                throw new EmptyDescriptionException(userInput[0]);
+            }
+            if (!mainTask.contains("/by")){
+                throw new EmptyDateException();
+            }
+            System.out.println(format);
+            String[] deadlineArray = mainTask.trim().split("/by", 2);
+            taskArrayList.add(new Deadline(deadlineArray[0],deadlineArray[1]));
+            //count = taskAdder(taskArrayList, count);
+            taskAdder(taskArrayList);
+            System.out.println(format);
+            try {
+                writeFile();
+            } catch (IOException | DukeException e) {
+                System.out.println("There was a problem saving your task! Please try again!");
+                System.out.println(format);
+            }
+        }
+
+        else if (userInput[0].equals("event")){ //ADDING AN EVENT
+            if (mainTask == null){
+                throw new EmptyDescriptionException(userInput[0]);
+            }
+            if (!mainTask.contains("/at")){
+                throw new EmptyDateException();
+            }
+            System.out.println(format);
+            String[] eventArray = mainTask.trim().split("/at", 2);
+            taskArrayList.add(new Event (eventArray[0], eventArray[1]));
+            //count = taskAdder(taskArrayList, count);
+            taskAdder(taskArrayList);
+            System.out.println(format);
+        }
+
+        else if (userInput[0].equals("delete")) {
+            if (mainTask == null) {
+                throw new EmptyDescriptionException(userInput[0]);
+            }
+            System.out.println(format);
+            //taskArrayList.remove(Integer.parseInt(userInput[1]) - 1);
+            //count = taskRemover(taskArrayList, Integer.parseInt(userInput[1]) - 1, count);
+            taskRemover(taskArrayList, Integer.parseInt(userInput[1]) - 1);
+            System.out.println(format);
+            try {
+                writeFile();
+            } catch (IOException | DukeException e) {
+                System.out.println("There was a problem saving your task! Please try again!");
+                System.out.println(format);
+            }
+        }
+
+        else {
+            throw new WrongCommandException();
+        }
+    }
```

Comment:
Try to break this long method into several lshort methods (as a  gereral rule no method should be more than 30 lines)